### PR TITLE
Pol 166 update google policies for credential service 1

### PR DESCRIPTION
**Pull request by gopinathmahakud1**
### Description

Replaced auth declaration with the credential declaration
Removed long_description field and use new info hash for Version, providers and service.
Updated readme to replace credential instructions with instructions to use credential service. 
Updated CWF definitions as needed.
Removed the Google Project param and used the Projects API to fetch projects details. 

### Issues Resolved

Replaced auth declaration with the credential declaration
Removed long_description field and use new info hash for Version, providers and service.
Updated readme to replace credential instructions with instructions to use credential service. 
Updated CWF definitions as needed.
Removed the Google Project param and used the Projects API to fetch projects details. 

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e3be9554ec7b30001fc4eb1
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e3beae74ec7b30001fc4eb6

### CODE REVIEW

File: compliance/google/long_stopped_instances/Projects 0.url [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[InternetShortcut]
```
Comment (cdwilhelm) [inline compliance/google/long_stopped_instances/Projects 0.url]: I think this file was added by mistake.

File: compliance/google/long_stopped_instances/README.md [hunk 2]
```diff
@@ -8,27 +8,37 @@ This policy checks all Google instances that are stopped and reports on any that
...
+
+### Credential configuration
+
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
+
+Provider tag value to match this policy: `gce`
+
+Required permissions in the provider:
+
+- The `Monitoring Viewer` Role, the `compute.instances.delete`, `compute.instances.list` Permissions
```
Comment (cdwilhelm) [inline compliance/google/long_stopped_instances/README.md]: add a bullet for each permission/role

File: compliance/google/long_stopped_instances/README.md [hunk 3]
```diff
@@ -8,27 +8,37 @@ This policy checks all Google instances that are stopped and reports on any that
...
+- Scope for the credential is <https://www.googleapis.com/auth/monitoring.write https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/monitoring.read https://www.googleapis.com/auth/monitoring https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/compute.readonly https://www.googleapis.com/auth/devstorage.full_control>
```
Comment (cdwilhelm) [inline compliance/google/long_stopped_instances/README.md]: scope isn't needed in the readme, that is included in the credential doc

File: cost/google/idle_compute_instances/README.md [hunk 4]
```diff
@@ -11,25 +11,41 @@ This Policy Template checks for idle instance in Google Compute Engine and then
...
+
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
+
+Provider tag value to match this policy: `gce`
+
+Required permissions in the provider:
 
-- Cloud Management - The `credential_viewer`,`observer` roles
 - Google - The `Monitoring Viewer` Role, and the `compute.instances.delete`, `compute.instances.list`, `compute.instances.get` Permissions
+- Scope for the credential is <https://www.googleapis.com/auth/monitoring.write https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/monitoring.read https://www.googleapis.com/auth/monitoring https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/compute.readonly https://www.googleapis.com/auth/devstorage.full_control>
```
Comment (cdwilhelm) [inline cost/google/idle_compute_instances/README.md]: No longer needed

File: cost/google/idle_compute_instances/README.md [hunk 5]
```diff
@@ -11,25 +11,41 @@ This Policy Template checks for idle instance in Google Compute Engine and then
...
+### Credential configuration
+
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
+
+Provider tag value to match this policy: `gce`
+
+Required permissions in the provider:
 
-- Cloud Management - The `credential_viewer`,`observer` roles
 - Google - The `Monitoring Viewer` Role, and the `compute.instances.delete`, `compute.instances.list`, `compute.instances.get` Permissions
```
Comment (cdwilhelm) [inline cost/google/idle_compute_instances/README.md]: please create bullet for each permission/role.

File: cost/google/instances_stackdriver_utilization/README.md [hunk 6]
```diff
@@ -36,6 +35,7 @@ Provider tag value to match this policy: `gce`
 Required permissions in the provider:
 
 - The `Monitoring Viewer` Role
+- Scope for the credential is <https://www.googleapis.com/auth/monitoring.write https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/monitoring.read https://www.googleapis.com/auth/monitoring https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/compute.readonly https://www.googleapis.com/auth/devstorage.full_control>
```
Comment (cdwilhelm) [inline cost/google/instances_stackdriver_utilization/README.md]: scope is no longer needed.

File: cost/google/instances_stackdriver_utilization/README.md [hunk 7]
```diff
@@ -36,6 +35,7 @@ Provider tag value to match this policy: `gce`
 Required permissions in the provider:
 
 - The `Monitoring Viewer` Role
```
Comment (cdwilhelm) [inline cost/google/instances_stackdriver_utilization/README.md]: I believe more permissions are needed.  At least a read and update type permission.

File: cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt [hunk 8]
```diff
@@ -368,10 +385,10 @@ policy "pol_utilization" do
     summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} rows containing Google instance StackDriver Utilization data"
     detail_template <<-EOS
 # Google Instance StackDriver Utilization
-| Region | instanceID | Hostname | Current Machine Type | Machine Type To Downsize | CPU Maximum % | CPU Minimum % | CPU Average % | Disk Maximum % | Disk Minimum % | Disk Average % | Memory Minimum % | Memory Maximum % | Memory Average % | Tags |
-| ------ | ---------- | -------- | -------------------- | ------------------------ | ------------- | ------------- | ------------- | -------------- | -------------- | -------------- | ---------------- | ---------------- | ---------------- | ---- |
+| Project ID | Region | instanceID | Hostname | Current Machine Type | Machine Type To Downsize | CPU Maximum % | CPU Minimum % | CPU Average % | Disk Maximum % | Disk Minimum % | Disk Average % | Memory Minimum % | Memory Maximum % | Memory Average % | Tags |
```
Comment (cdwilhelm) [inline cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt]: Sort by projectid then region.

File: cost/google/object_storage_optimization/README.md [hunk 9]
```diff
@@ -1,48 +1,54 @@
...
+
+Provider tag value to match this policy: `gce`
+
+Required permissions in the provider:
 
 - This policy requires IAM permissions to storage.buckets.list, storage.buckets.getIamPolicy2, storage.objects.list, storage.objects.getIamPolicy2,6, storage.objects.create (for the destination bucket), storage.objects.delete (for the destination bucket)4, storage.objects.get (for the source bucket) and storage.objects.delete.
 - Create a service account (if not exists) with the necessary permissions under Google-cloud platform (IAM & admin -> service accounts). Generate key, a JSON file will get downloaded in which you can find 'client email' and 'private key' which has to be added as credentials in RightScale cloud management Design -> Credentials with name 'GCE_PLUGIN_ACCOUNT' and 'GCE_PLUGIN_PRIVATE_KEY' respectively.
+- Scope for the credential is <https://www.googleapis.com/auth/monitoring.write https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/monitoring.read https://www.googleapis.com/auth/monitoring https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/compute.readonly https://www.googleapis.com/auth/devstorage.full_control>
```
Comment (cdwilhelm) [inline cost/google/object_storage_optimization/README.md]: scope is no longer needed.

File: cost/google/object_storage_optimization/README.md [hunk 10]
```diff
@@ -1,48 +1,54 @@
...
+### Credential configuration
+
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
+
+Provider tag value to match this policy: `gce`
+
+Required permissions in the provider:
 
 - This policy requires IAM permissions to storage.buckets.list, storage.buckets.getIamPolicy2, storage.objects.list, storage.objects.getIamPolicy2,6, storage.objects.create (for the destination bucket), storage.objects.delete (for the destination bucket)4, storage.objects.get (for the source bucket) and storage.objects.delete.
 - Create a service account (if not exists) with the necessary permissions under Google-cloud platform (IAM & admin -> service accounts). Generate key, a JSON file will get downloaded in which you can find 'client email' and 'private key' which has to be added as credentials in RightScale cloud management Design -> Credentials with name 'GCE_PLUGIN_ACCOUNT' and 'GCE_PLUGIN_PRIVATE_KEY' respectively.
```
Comment (cdwilhelm) [inline cost/google/object_storage_optimization/README.md]: This is covered in the credential docs.

File: cost/google/object_storage_optimization/README.md [hunk 11]
```diff
@@ -1,48 +1,54 @@
...
+
+### Credential configuration
+
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
+
+Provider tag value to match this policy: `gce`
+
+Required permissions in the provider:
 
 - This policy requires IAM permissions to storage.buckets.list, storage.buckets.getIamPolicy2, storage.objects.list, storage.objects.getIamPolicy2,6, storage.objects.create (for the destination bucket), storage.objects.delete (for the destination bucket)4, storage.objects.get (for the source bucket) and storage.objects.delete.
```
Comment (cdwilhelm) [inline cost/google/object_storage_optimization/README.md]: please list each role/permission in a bullet for easier reading.

File: cost/google/object_storage_optimization/google_object_storage_optimization.pt [hunk 12]
```diff
@@ -226,10 +243,10 @@ policy "google_object_storage_optimization" do
     summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): Google Object Storage Optimization."
     detail_template <<-EOS
 # Found {{ len data }} Google bucket objects which are last modified outside of the specified timeframe.
-| Bucket Name | Region | Object Name | Current Storage Class | Last Modified Date | Move Storage Class To | Tags |
-| ----------- | ------ | ----------- | --------------------- | ------------------ | --------------------- | ---- |
+| Project ID | Bucket Name | Region | Object Name | Current Storage Class | Last Modified Date | Move Storage Class To | Tags |
```
Comment (cdwilhelm) [inline cost/google/object_storage_optimization/google_object_storage_optimization.pt]: please sort by ProjectId then region.

File: security/storage/google/public_buckets/README.md [hunk 13]
```diff
@@ -1,38 +1,41 @@
...
-### Google Required Permissions
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
+
+Provider tag value to match this policy: `gce`
+
+Required permissions in the provider:
 
 - storage.buckets.list
 - storage.buckets.getIamPolicy2
+- Scope for the credential is <https://www.googleapis.com/auth/monitoring.write https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/monitoring.read https://www.googleapis.com/auth/monitoring https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/compute.readonly https://www.googleapis.com/auth/devstorage.full_control>
```
Comment (cdwilhelm) [inline security/storage/google/public_buckets/README.md]: scope is no longer needed.

File: cost/google/idle_compute_instances/google_idle_compute_instances.pt [hunk 14]
```diff
@@ -262,10 +289,10 @@ policy "pol_utilization" do
     summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} rows containing Google instance StackDriver Utilization data"
     detail_template <<-EOS
 # Google Instance StackDriver Utilization
-| Region | instanceID | Hostname | CPU Maximum % | CPU Minimum % | CPU Average % | Memory Minimum % | Memory Maximum % | Memory Average % | Labels |
-| ------ | ---------- | -------- | ------------- | ------------- | ------------- | ---------------- | ---------------- | ---------------- | ------ |
+| Project ID | Region | instanceID | Hostname | CPU Maximum % | CPU Minimum % | CPU Average % | Memory Minimum % | Memory Maximum % | Memory Average % | Labels |
```
Comment (cdwilhelm) [inline cost/google/idle_compute_instances/google_idle_compute_instances.pt]: please sort by projectid and region

File: compliance/google/long_stopped_instances/google_long_stopped_instances.pt [hunk 15]
```diff
@@ -197,10 +228,10 @@ policy "pol_utilization" do
     summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} google cloud instances in Stopped State for Over {{parameters.param_stoped_days}} Day(s)"
     detail_template <<-EOS
 # Google Instance Out of Compliance
-|Instance ID| Name | Status | Zone | Tags |
-|:---------:|-----:|-------:|-----:|-----:|
+| Project ID | Instance ID | Name | Status | Zone | Tags |
```
Comment (cdwilhelm) [inline compliance/google/long_stopped_instances/google_long_stopped_instances.pt]: please sort by project id and region/zone.  rename zone to regions and mov next to projectid.

File: compliance/google/long_stopped_instances/Projects 0.url (hunk 1 above)
Comment (gopinathmahakud1) [reply compliance/google/long_stopped_instances/Projects 0.url]: Yes Curt I have added by mistake. I have deleted this file

File: compliance/google/long_stopped_instances/README.md (hunk 2 above)
Comment (gopinathmahakud1) [reply compliance/google/long_stopped_instances/README.md]: I have addressed this comment please review the changes.

File: compliance/google/long_stopped_instances/README.md (hunk 3 above)
Comment (gopinathmahakud1) [reply compliance/google/long_stopped_instances/README.md]: I have addressed this comment please review the changes.

File: cost/google/idle_compute_instances/README.md (hunk 4 above)
Comment (gopinathmahakud1) [reply cost/google/idle_compute_instances/README.md]: I have addressed this comment please review the changes.

File: cost/google/idle_compute_instances/README.md (hunk 5 above)
Comment (gopinathmahakud1) [reply cost/google/idle_compute_instances/README.md]: I have addressed this comment please review the changes.

File: cost/google/instances_stackdriver_utilization/README.md (hunk 6 above)
Comment (gopinathmahakud1) [reply cost/google/instances_stackdriver_utilization/README.md]: I have addressed this comment please review the changes.

File: cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt (hunk 8 above)
Comment (gopinathmahakud1) [reply cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt]: I have addressed this comment please review the changes.

File: cost/google/object_storage_optimization/README.md (hunk 9 above)
Comment (gopinathmahakud1) [reply cost/google/object_storage_optimization/README.md]: I have addressed this comment please review the changes.

File: cost/google/object_storage_optimization/README.md (hunk 10 above)
Comment (gopinathmahakud1) [reply cost/google/object_storage_optimization/README.md]: I have addressed this comment please review the changes.

File: cost/google/object_storage_optimization/README.md (hunk 11 above)
Comment (gopinathmahakud1) [reply cost/google/object_storage_optimization/README.md]: I have addressed this comment please review the changes.

File: cost/google/object_storage_optimization/google_object_storage_optimization.pt (hunk 12 above)
Comment (gopinathmahakud1) [reply cost/google/object_storage_optimization/google_object_storage_optimization.pt]: I have addressed this comment please review the changes.

File: security/storage/google/public_buckets/README.md (hunk 13 above)
Comment (gopinathmahakud1) [reply security/storage/google/public_buckets/README.md]: I have addressed this comment please review the changes.

File: cost/google/idle_compute_instances/google_idle_compute_instances.pt (hunk 14 above)
Comment (gopinathmahakud1) [reply cost/google/idle_compute_instances/google_idle_compute_instances.pt]: I have addressed this comment please review the changes.

File: compliance/google/long_stopped_instances/google_long_stopped_instances.pt (hunk 15 above)
Comment (gopinathmahakud1) [reply compliance/google/long_stopped_instances/google_long_stopped_instances.pt]: I have addressed this comment please review the changes.

File: cost/google/instances_stackdriver_utilization/README.md (hunk 7 above)
Comment (gopinathmahakud1) [reply cost/google/instances_stackdriver_utilization/README.md]: I have addressed this comment please review the changes.

**Pull request merged**